.github/workflows/reusable-workflow-sast.yml

```diff
@@ -75,9 +79,79 @@ on:
         required: false
 
 jobs:
+  test-visibility:
+
+    name: Test repo visibility
+    runs-on: ubuntu-latest
+
+    outputs:
+      visibility: ${{ steps.check-visibility.outputs.visibility }}
+
+    steps:
+
+      - name: Check repo visibility (GHAS currently not procured)
+        id: check-visibility
+        run: |
+          status_code=$(curl -s -o /dev/null -w "%{http_code}" https://github.com/${{ github.repository_owner }}/${{ github.event.repository.name }})
+          if [ $status_code -eq 200 ];
+          then
+            echo $status_code
+            visibility=public
+            echo "visibility=public" >> "$GITHUB_OUTPUT"
+          else
+            echo $status_code
+            echo "visibility=private" >> "$GITHUB_OUTPUT"
+            visibility=private
+          fi
+
+          echo ${{ github.actor }}
+          echo ${{ inputs.force_semgrep }}
+          echo $visibility
+          echo ${{ inputs.force_semgrep }}
+
+  semgrep:
+    name: semgrep-oss/scan
+    runs-on: ubuntu-latest
+
+    needs: test-visibility
+
+    permissions:
+      actions: read
+      contents: read
+      pull-requests: read
+
+    container:
+      # A Docker image with Semgrep installed. Do not change this.
+      image: semgrep/semgrep
+
+    if: ${{ (github.actor != 'dependabot[bot]') && (inputs.force_semgrep == true || needs.test-visibility.outputs.visibility == 'private') }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      # Run the "semgrep scan" command on the command line of the docker image.
+      - run: semgrep scan --config auto --sarif --sarif-output=semgrep.sarif --force-color
+
+      - name: Name sarif outputs
+        id: name-sarifs
+        run: |
+          service=$(echo ${{ github.repository_owner }}_${{ github.event.repository.name }} | tr '[:upper:]' '[:lower:]' | sed -e 's|/|_|g' -e 's|\.|_|g' -e 's| |_|g' -e 's|-|_|g')
```

Comment:
Can you add a comment explaining what this line does? It's a bit obscure

Reply:
Comment added above the run, good point 👍🏻